nixos/oci-image: expose diskSize as a configurable option

siriobalmelli opened the pull request:
diskSize defaults to the previous hard-coded 8192: no change for existing users.

Users can set diskSize when building images which require larger disk space; thus avoiding the error:

    ERROR: cptofs failed. diskSize might be too small for closure.

## Description of changes

- add `diskSize` to `nixos/modules/virtualisation/oci-options.nix`
- reference `diskSize` in `nixos/modules/virtualisation/oci-image.nix`

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [N/A] x86_64-darwin
  - [N/A] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [N/A] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [N/A] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [N/A] (Package updates) Added a release notes entry if the change is major or breaking
  - [N/A] (Module updates) Added a release notes entry if the change is significant
  - [N/A] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

superherointj commented on nixos/modules/virtualisation/oci-options.nix:
```diff
@@ -9,6 +9,13 @@
+        type = lib.types.int;
+        default = 8192;
+        description = ''
+          Size of the disk image created.
```
Would be good to mention the unit used. And an example that a human can properly understand.

siriobalmelli replied:
Good point. Respun.

superherointj commented on nixos/modules/virtualisation/oci-options.nix:
```diff
@@ -9,6 +9,12 @@
+        type = lib.types.int;
+        default = 8192;
+        description = "Size of the disk image created in MB.";
+        example = "diskSize = 12288; # 12GiB";
```
```suggestion
        example = "diskSize = 12 * 1024; # 12GiB";
```
Less hard for humans. They're bad at division. Monkey patching is easier.

siriobalmelli replied:
Good suggestion. Respun, added you as co-author.

superherointj commented:
I need you to reword commit title to:

> nixos/oci-image: expose diskSize as a configurable option

Because you are changing a NixOS module.

siriobalmelli committed:
nixos/oci-image: expose diskSize as a configurable option

diskSize defaults to the previous hard-coded 8192:
no change for existing users.

Users can set diskSize when building images which require
larger disk space; thus avoiding the error:

    ERROR: cptofs failed. diskSize might be too small for closure.

Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>
Co-authored-by: superherointj <5861043+superherointj@users.noreply.github.com>
